Linking issue since 5.1-1.8.0

**Host Environment**
- OS: Linux x64

**Failure logs**
```text
/usr/bin/ld: ../vcpkg_installed/x64-linux/debug/lib/libzippp_static.a(libzippp.cpp.o): warning: relocation against `_ZSt7nothrow@@GLIBCXX_3.4' in read-only section `.text'
/usr/bin/ld: ../vcpkg_installed/x64-linux/debug/lib/libzippp_static.a(libzippp.cpp.o): relocation R_X86_64_PC32 against symbol `_ZTVN8libzippp8ZipEntryE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

See https://github.com/Werni2A/OpenAllegroParser/runs/7693726908 for the full compile log.

**Additional context**
I tried to figure out which commit in vcpkg introduced the issue by fixing the baseline to a few 'interesting' commits. It happens that after incorporating https://github.com/microsoft/vcpkg/pull/24885 my software does not link anymore. The major change was to bump the libzippp package, therefore I guess that the issue is located somewhere here.

Do you experience similar issues or am I doing something wrong?

Hi, I tested again with `6.0-1.9.2#1` from vcpkg but the same issue still persists.

I tried adding `set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)` to my CMakeFile which changes the the linking issue slightly. Find the error log below. Interestingly the PIC flag changes the linking issue against `_ZTVN8libzippp8ZipEntryE` as reported initially to `stderr@@GLIBC_2.2.5`. Still I have no idea what this issue is about. I don't even know if it's related to this library, the vcpkg port files or my own CMake scripts.

In case you have no suggestions what to do feel free to close this issue. I guess there is low value in leaving this open. Looks like I'm the only one having this issue anyway.

```text
/usr/bin/ld: ../vcpkg_installed/x64-linux/debug/lib/libzippp_static.a(libzippp.cpp.o): warning: relocation against `_ZSt7nothrow@@GLIBCXX_3.4' in read-only section `.text'
/usr/bin/ld: ../vcpkg_installed/x64-linux/debug/lib/libzippp_static.a(libzippp.cpp.o): relocation R_X86_64_PC32 against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,9 +25,9 @@ if(NOT CMAKE_BUILD_TYPE)
 endif()
 
 if(NOT WIN32)
-  set(CMAKE_CXX_FLAGS "-W -Wall -Wextra -ansi -pedantic -std=c++11")
-  set(CMAKE_CXX_FLAGS_DEBUG "-g")
-  set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -W -Wall -Wextra -ansi -pedantic -std=c++11")
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g")
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
 endif()
 
 option(LIBZIPPP_INSTALL "Install library" ${is_root_project})

